protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,137 @@
+        address _originator, 
+        address _recipient, 
+        bytes32 _hashLock, 
+        uint _timeLock
+    ) external payable override returns (bytes32 _bridgeTransferId) {
+        console.log("initiateBridgeTransfer");
+        uint256 totalAmount = _wethAmount;
+
+        // If msg.value is greater than 0, convert ETH to WETH and add to total amount
+        if (msg.value > 0) {
+            weth.deposit{value: msg.value}();
+            //Transfer WETH to this contract, revert if transfer fails
+            require(weth.transfer(address(this), msg.value), "WETH transfer failed");
+            totalAmount += msg.value;
+        }
+
+        // Ensure there is a valid total amount
+        require(totalAmount > 0, "Total amount must be greater than 0");
+
+        _bridgeTransferId = keccak256(
+            abi.encodePacked(
+                _originator, 
+                _recipient, 
+                _hashLock, 
+                _timeLock, 
+                block.timestamp
+        ));
+
+        // Check if the bridge transfer already exists
+        require(bridgeTransfers[_bridgeTransferId].amount == 0, "Bridge transfer already exists");
+
+        bridgeTransfers[_bridgeTransferId] = BridgeTransfer({
+            amount: totalAmount,
+            originator: _originator,
+            recipient: _recipient,
+            hashLock: _hashLock,
+            timeLock: block.timestamp + _timeLock,
+            exists: true
+        });
+
+        emit BridgeTransferInitiated(_bridgeTransferId, _originator, _recipient, _hashLock, _timeLock);
+        return _bridgeTransferId;
+    }
+
+    function completeBridgeTransfer(bytes32 _bridgeTransferId, bytes32 _secret) external override {
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[_bridgeTransferId];
+        require(bridgeTransfer.exists, "Bridge transfer does not exist");
+        require(keccak256(abi.encodePacked(_secret)) == bridgeTransfer.hashLock, "Invalid secret");
+
+        // Move the bridge transfer to completed
+        completedBridgeTransfers[_bridgeTransferId] = bridgeTransfer;
+
+        // Delete from active bridgeTransfers
+        delete bridgeTransfers[_bridgeTransferId];
+
+        payable(bridgeTransfer.recipient).transfer(bridgeTransfer.amount);
+
+        emit BridgeTransferCompleted(_bridgeTransferId, _secret);
+    }
+
+    function refundBridgeTransfer(bytes32 _bridgeTransferId) external override {
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[_bridgeTransferId];
+        require(bridgeTransfer.exists, "Bridge transfer does not exist");
+        require(block.timestamp > bridgeTransfer.timeLock, "Timelock has not expired");
+
+        delete bridgeTransfers[_bridgeTransferId];
+
+        payable(bridgeTransfer.originator).transfer(bridgeTransfer.amount);
+
+        emit BridgeTransferRefunded(_bridgeTransferId);
+    }
+
+    function getBridgeTransferDetail(bytes32 _bridgeTransferId) external view override returns (
```

Comment:
we get this out of the box I think so seems like we can remove this entire function